### PR TITLE
Cleanup deps

### PR DESCRIPTION
**Pull request by Ericson2314**
- `nativeBuildInputs` vs `buildInputs`

- narrow down `with`s for clarity

- use `autoreconfHook` not `bootstrap` script

These sorts of changes have also been done in the Nix repo.

### CODE REVIEW

File: flake.nix [hunk 1]
```diff
@@ -141,15 +142,20 @@
 
           src = self;
 
-          buildInputs =
-            [
+          nativeBuildInputs =
+            with final.buildPackages; [
```
Comment (SuperSandro2000) [inline flake.nix:146]:
```suggestion
          nativeBuildInputs = with final.buildPackages; [
```

File: flake.nix [hunk 2]
```diff
@@ -141,15 +142,20 @@
...
               libtool
-              unzip
               nukeReferences
               pkg-config
+              mdbook
+            ];
+
+          buildInputs =
+            with final; [
+              unzip
```
Comment (SuperSandro2000) [inline flake.nix:158]:
```suggestion
          buildInputs = with final; [
            unzip
```

File: flake.nix (hunk 1 above)
Comment (Ericson2314) [reply flake.nix:146]: I don't like the existing style either, but if I did that I would be better off moving way more things around?

Comment (SuperSandro2000) [reply flake.nix:146]: hmmm but that would also be a much bigger change and I didn't see the other lines in the diff 😉

**Pull request merged**